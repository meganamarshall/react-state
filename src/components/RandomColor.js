import React, { PureComponent } from 'react';

export default class RandomColor extends PureComponent {
  state = {
    color: '',
    trigger: new Date(),
    img: ''
  }

  randomColor = () => {
    const myColors = ['red', 'blue', 'blueviolet', 'yellow', 'yellowgreen', 'coral', 'crimson', 'cornflowerblue'];
    const i = Math.floor(Math.random() * myColors.length);

    return myColors[i];
  }
  
  componentDidMount() {
    this.intervalId = setInterval(() => {
      this.setState({ color: this.randomColor(), trigger: new Date() });
    }, 1500);
  }
  
  componentDidUpdate(prevProps, prevState) {
    if(prevState.color === this.state.color) {
      this.setState({ img: 'https://i.dailymail.co.uk/i/pix/2016/03/18/15/324D202500000578-3498922-image-a-33_1458315465874.jpg' });
    }
  }

  componentWillUnmount() {
    clearInterval(this.intervalId);
  }
  
  render() {
    const { color, img } = this.state;
    const styles = {
      background: img ? `center / contain url(${img})` : color,
      width: '100px',
      height: '100px',
      border: '1px solid black'
    };
    return <div style={styles}></div>;
  }
}
