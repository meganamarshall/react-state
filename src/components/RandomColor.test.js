import React from 'react';
import { shallow } from 'enzyme';
import RandomColor from './RandomColor';

describe('randomcolor component', () => {
  it('renders the random color', () => {
    const wrapper = shallow(<RandomColor />);
    expect(wrapper).toMatchSnapshot();
  });
});
