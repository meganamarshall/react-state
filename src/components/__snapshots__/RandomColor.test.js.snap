// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`randomcolor component renders the random color 1`] = `
<div
  style={
    Object {
      "background": "",
      "border": "1px solid black",
      "height": "100px",
      "width": "100px",
    }
  }
/>
`;
